### DISCUSSION TITLE
Xamarin forms to .net maui

### DISCUSSION THREAD

**Discussion by AzhanMemon** [dotnet/maui#20594] [Upgrading from Xamarin]
I have a custom renderers which convert native android linear layout view to xamairn view so now when migrating to .net maui how shall i achieve this

**Comment by davidortinau** [comment]
I would try the custom renderer first like this https://learn.microsoft.com/en-us/dotnet/maui/migration/custom-renderers?view=net-maui-8.0.

Have you considered [FlexLayout](https://learn.microsoft.com/dotnet/maui/user-interface/layouts/flexlayout?view=net-maui-8.0) for this purpose? Maybe you could reduce the code you need to maintain.

If the above doesn't immediately work, you are probably better off porting to a handler like this https://learn.microsoft.com/dotnet/maui/migration/renderer-to-handler?view=net-maui-8.0.

I have a couple of ported handlers in this project, though they may not be great examples if your renderer is primarily doing layout. https://github.com/davidortinau/SmartHotel360-Mobile/blob/main/Documents/UpgradeDiary/UpgradingSmartHotel360.md
